packages/core/entity/FileDetail/index.ts

```diff
@@ -126,56 +144,74 @@ export default class FileDetail {
         // staging.files.allencell.org/130/b23/bfe/117/2a4/71b/746/002/064/db4/1a/danny_int_test_4.txt
-        if (typeof path === "string" && path.startsWith(AICS_FMS_S3_BUCKET)) {
+        if (
+            typeof path === "string" &&
+            path.startsWith(AICS_FMS_S3_BUCKETS[this.env]) &&
+            AICS_FMS_S3_BUCKETS[this.env]
+        ) {
             return FileDetail.convertAicsS3PathToHttpUrl(path) as string;
         }
 
         // Otherwise just return the path as is and hope for the best
         return path as string;
     }
 
+    public get cloudPath(): string {
+        //// REMOVE THIS (BACKWARDS COMPAT)
+        if (this.path.startsWith("/allen")) {
+            return this.convertAicsDrivePathToAicsS3Path(this.path);
+        }
+        ////
+
+        // AICS FMS files' paths are cloud paths
+        return this.path;
+    }
+
     public get localPath(): string | null {
-        // REMOVE THIS (BACKWARDS COMPAT)
+        //// REMOVE THIS (BACKWARDS COMPAT)
         if (this.path.startsWith("/allen")) {
             return this.path;
         }
+        ////
 
-        const localPath = this.getFirstAnnotationValue("Local File Path");
-        if (localPath === undefined) {
-            return null;
+        if (this.getAnnotation("Cache Eviction Date")) {
+            return this.getLocalPath();
         }
-        return localPath as string;
+        return null;
     }
 
-    public get cloudPath(): string {
-        // REMOVE THIS (BACKWARDS COMPAT)
-        if (this.path.startsWith("/allen")) {
-            return FileDetail.convertAicsDrivePathToAicsS3Path(this.path);
+    public getLocalPath(): string | null {
+        const localPrefix = HOST_PREFIXES[this.env];
+        const cloudPrefix = `${AICS_FMS_S3_URL_PREFIX}${AICS_FMS_S3_BUCKETS[this.env]}`;
+
+        if (this.path.startsWith(cloudPrefix)) {
+            const relativePath = this.path.replace(`${cloudPrefix}`, "");
+            return `${localPrefix}${relativePath}`;
         }
 
-        // AICS FMS files' paths are cloud paths
         return this.path;
     }
 
     public get downloadPath(): string {
-        // REMOVE THIS (BACKWARDS COMPAT)
+        //// REMOVE THIS (BACKWARDS COMPAT)
         if (this.path.startsWith("/allen")) {
             return `http://aics.corp.alleninstitute.org/labkey/fmsfiles/image${this.path}`;
         }
+        ////
 
+        const localPath = this.getLocalPath();
         // For AICS files that are available on the Vast, users can use the cloud path, but the
         // download will be faster and not incur egress fees if we download via the local network.
-        if (this.localPath && this.localPath.startsWith("/allen")) {
-            return `http://aics.corp.alleninstitute.org/labkey/fmsfiles/image${this.localPath}`;
+        if (localPath && localPath.startsWith("/allen") && this.env === Environment.PRODUCTION) {
+            return `http://aics.corp.alleninstitute.org/labkey/fmsfiles/image${localPath}`;
```

Comment:
Perhaps if you are introducing env dependent variables to this class then there should be a case for `this.env === Environment.STAGING` where the return is:
`return `http://stg-aics.corp.alleninstitute.org/labkey/fmsfiles/image${localPath}`;`

Reply:
We dont have labkey download for staging files I believe so this function default returns the cloud path for the staging file which can be downloaded